app/interface/deps/usecases/course_deps.py

```
@@ -0,0 +1,32 @@
+    CourseListRequest,
+    AllCoursesRequest,
+)
+from app.application.usecases.course import (
+    ActivityListUseCase,
+    CourseListUseCase,
+    AllCoursesUseCase,
+)
+
+
+def get_activity_list_usecase(
+    scraper: LmsScraper = Depends(get_lms_scraper, use_cache=False),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)